operator/internal/handlers/lokistack_enable_zone_awareness.go

```diff
@@ -0,0 +1,171 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
+	"github.com/grafana/loki/operator/internal/manifests"
+)
+
+const (
+	LokiInstanceAZ = "loki_instance_availability_zone"
```

Comment:
After testing this PR a bit on a playground cluster, I noticed that annotations managed by the Loki Operator follow this pattern: `loki.grafana.com/label-name`. I suggest to rename this `loki.grafana.com/availability-zone`. 

nit. Also let's follow the var naming pattern for Annotation variables like in `manifests/var.go`: `annotationLokiInstanceAZ`